astrosat/templates/astrosat/admin/date_range_filter.html

```diff
@@ -0,0 +1,94 @@
+{% load i18n %}
+{% load static %}
+
+<link rel="stylesheet" type="text/css" href="{% static 'admin/css/widgets.css' %}">
+<style>
+    .date-range-filter .button, .date-range-filter input[type=submit], .date-range-filter input[type=reset] {
+        /* make all the buttons the same size */
+        padding: 4px 5px;
+        border: none;
+        border-radius: 4px;
+    }
+    .date-range-filter {
+        /* indent stuff horizontally */
+        padding-left: 15px;
+    }
+    .date-range-filter p {
+        /* don't indent stuff vertically */
+        padding-left: 0px;
+        line-height: 0;
+    }
+    .date-range-filter .datetimeshortcuts {
+        /* hide text & float icons */
+        font-size: 0;
+        float: right;
+        position: absolute;
+        padding-top: 4px;
+    }
+    .date-range-filter a {
+        /* pad stuff a bit */
+        position: absolute;
+        padding-top: 3px;
+        padding-left: 4px;
+    }
+    @media (min-width: 768px) {
+        .calendarbox {
+            margin-left: -16em !important;
+            margin-top: 9em !important;
+        }
+    }
+    @media (max-width: 767px) {
+        .calendarbox {
+            overflow: visible;
+        }
+    }
+</style>
+
+{# the admin has (a namespaced version of) jQuery built-in #}
+{# as per: https://docs.djangoproject.com/en/3.1/ref/contrib/admin/#jquery #}
+
+<script>
+
+    // only import and init DateTimeShortcuts once regardless of how many times
+    // this template is used in an Admin; had to pre-compute the static paths b/c... Django
+
+    django.jQuery('document').ready(function () {
+        if (!('DateTimeShortcuts' in window)) {
+            django.jQuery.when(
+                {% for script in spec.form.scripts %}
+                    django.jQuery.getScript('{{script}}'),
+                {% endfor %}
+            ).then(function(){
+                django.jQuery('.datetimeshortcuts').remove();
+                DateTimeShortcuts.init();
+            });
+        }
+    });
+
+    function submit_date_range_filter(event, choice_input_id, form_id) {
+        event.preventDefault();
+        form_data = django.jQuery("#" + form_id).serialize();
+        query_string = django.jQuery("#" + choice_input_id).val();
+        window.location = window.location.pathname + query_string + "&" + form_data;
+    }
+
+    function reset_date_range_filter(choice_input_id) {
+        query_string = django.jQuery("#" + choice_input_id).val();
+        window.location = window.location.pathname + query_string;
+    }
+
+</script>
```

Comment:
I'd export this JS into a separate file as well, but same as above, if you prefer this, it won't prevent PR approval.

Reply:
I can't (easily) move the JS to an external file b/c it includes Django Template Tags.